diff --git a/src/putty/windows/winmiscs.c b/src/putty/windows/winmiscs.c
index 73e4f86..567d531 100644
--- a/src/putty/windows/winmiscs.c
+++ b/src/putty/windows/winmiscs.c
@@ -253,17 +253,29 @@ uintmax_t strtoumax(const char *nptr, char **endptr, int base)
 
 bool platform_aes_hw_available(void)
 {
-    return IsProcessorFeaturePresent(PF_ARM_V8_CRYPTO_INSTRUCTIONS_AVAILABLE);
+    #ifdef PF_ARM_V8_CRYPTO_INSTRUCTIONS_AVAILABLE
+        return IsProcessorFeaturePresent(PF_ARM_V8_CRYPTO_INSTRUCTIONS_AVAILABLE);
+    #else
+        return false;
+    #endif
 }
 
 bool platform_sha256_hw_available(void)
 {
-    return IsProcessorFeaturePresent(PF_ARM_V8_CRYPTO_INSTRUCTIONS_AVAILABLE);
+    #ifdef PF_ARM_V8_CRYPTO_INSTRUCTIONS_AVAILABLE
+        return IsProcessorFeaturePresent(PF_ARM_V8_CRYPTO_INSTRUCTIONS_AVAILABLE);
+    #else
+        return false;
+    #endif
 }
 
 bool platform_sha1_hw_available(void)
 {
-    return IsProcessorFeaturePresent(PF_ARM_V8_CRYPTO_INSTRUCTIONS_AVAILABLE);
+    #ifdef PF_ARM_V8_CRYPTO_INSTRUCTIONS_AVAILABLE
+        return IsProcessorFeaturePresent(PF_ARM_V8_CRYPTO_INSTRUCTIONS_AVAILABLE);
+    #else
+        return false;
+    #endif
 }
 
 #endif
