diff --git a/configure.ac b/configure.ac
index 70884db..3e7d6cb 100644
--- a/configure.ac
+++ b/configure.ac
@@ -413,10 +413,16 @@ if test "$buildmain" = "yes"; then
     makensisscript="yes"
     if echo $host_cpu | grep "x86_64" > /dev/null 2>&1; then
       NSIS_64BIT=1
+      NSIS_ARM64=0
+    elif echo $host_cpu | grep "aarch64" > /dev/null 2>&1; then
+      NSIS_64BIT=1
+      NSIS_ARM64=1
     else
       NSIS_64BIT=0
+      NSIS_ARM64=0
     fi
     AC_SUBST(NSIS_64BIT)
+    AC_SUBST(NSIS_ARM64)
   else
     makensisscript="no"
   fi
