From 014c7ced7cab3acb5dab8c7d4123b0ea955745c4 Mon Sep 17 00:00:00 2001
From: GH Cao <driver1998@foxmail.com>
Date: Thu, 20 Feb 2020 20:33:05 +0800
Subject: [PATCH 3/3] Enable shellext on Windows ARM

---
 configure.ac | 13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 30dfb78..58d92d4 100644
--- a/configure.ac
+++ b/configure.ac
@@ -127,9 +127,18 @@ AC_ARG_ENABLE(shellext, [], \
 
 if test "$localesonly" != "yes" && echo $host_os | grep "cygwin\|mingw" > /dev/null 2>&1 && test "$shellext" = "yes"; then
   cd "$OLD_DIR/src/fzshellext/32" || exit 1
-  "$ac_abs_confdir/src/fzshellext/configure" --prefix="$prefix" --exec-prefix="$exec_prefix" --host=i686-w64-mingw32 || exit 1
+  if echo $host_cpu | grep "armv7" > /dev/null 2>&1; then
+    "$ac_abs_confdir/src/fzshellext/configure" --prefix="$prefix" --exec-prefix="$exec_prefix" --host=armv7-w64-mingw32 || exit 1
+  else
+    "$ac_abs_confdir/src/fzshellext/configure" --prefix="$prefix" --exec-prefix="$exec_prefix" --host=i686-w64-mingw32 || exit 1
+  fi
+
   cd "$OLD_DIR/src/fzshellext/64" || exit 1
-  "$ac_abs_confdir/src/fzshellext/configure" --prefix="$prefix" --exec-prefix="$exec_prefix" --host=x86_64-w64-mingw32 || exit 1
+  if echo $host_cpu | grep "aarch64" > /dev/null 2>&1; then
+    "$ac_abs_confdir/src/fzshellext/configure" --prefix="$prefix" --exec-prefix="$exec_prefix" --host=aarch64-w64-mingw32 || exit 1
+  else
+    "$ac_abs_confdir/src/fzshellext/configure" --prefix="$prefix" --exec-prefix="$exec_prefix" --host=x86_64-w64-mingw32 || exit 1
+  fi
 else
   cd "$OLD_DIR/src/fzshellext/64" || exit 1
   "$ac_abs_confdir/src/fzshellext/configure" --disable-shellext || exit 1
-- 
2.17.1

